packages/blade/src/components/FileUpload/__tests__/FileUpload.test.stories.tsx

```diff
@@ -0,0 +1,449 @@
+  const { getByText } = within(document.body);
+
+  // Create a file
+  const blob = new Blob([
+    'data:image/png;base64,R0lGODlhDwAPAKECAAAAzMzM/////wAAACwAAAAADwAPAAACIISPeQHsrZ5ModrLlN48CXF8m2iQ3YmmKqVlRtW4MLwWACH+H09wdGltaXplZCBieSBVbGVhZCBTbWFydFNhdmVyIQAAOw==',
+  ]);
+  const filename = 'my-image.png';
+  const file = new File([blob], filename, {
+    type: 'image/png',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).not.toHaveAttribute('multiple');
+  await expect(input).toHaveAttribute('name', 'single-file-upload-input');
+
+  await userEvent.upload(input as HTMLElement, file);
+  await expect(getByText(filename)).toBeInTheDocument();
+};
+
+export const TestMultipleFileUpload: StoryFn<typeof Toast> = (): React.ReactElement => {
+  return (
+    <Box maxWidth="400px">
+      <FileUpload
+        uploadType="multiple"
+        label="Upload GST certificate"
+        helpText=" .jpg, .jpeg, or .png file only"
+        accept="image/*"
+        isRequired
+        necessityIndicator="required"
+        name="multiple-file-upload-input"
+      />
+    </Box>
+  );
+};
+
+TestMultipleFileUpload.play = async () => {
+  const { getByText } = within(document.body);
+
+  // Create files
+  const blob = new Blob([
+    'data:image/png;base64,R0lGODlhDwAPAKECAAAAzMzM/////wAAACwAAAAADwAPAAACIISPeQHsrZ5ModrLlN48CXF8m2iQ3YmmKqVlRtW4MLwWACH+H09wdGltaXplZCBieSBVbGVhZCBTbWFydFNhdmVyIQAAOw==',
+  ]);
+  const file1 = new File([blob], 'my-image.png', {
+    type: 'image/png',
+  });
+  const file2 = new File([blob], 'my-image2.png', {
+    type: 'image/png',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).toHaveAttribute('multiple');
+  await userEvent.upload(input as HTMLElement, [file1, file2]);
+  await expect(getByText('my-image.png')).toBeInTheDocument();
+  await expect(getByText('my-image2.png')).toBeInTheDocument();
+};
+
+export const TestOnRemove: StoryFn<typeof Toast> = (): React.ReactElement => {
+  return (
+    <Box maxWidth="400px">
+      <FileUpload
+        uploadType="single"
+        label="Upload GST certificate"
+        helpText=" .jpg, .jpeg, or .png file only"
+        accept="image/*"
+        isRequired
+        necessityIndicator="required"
+        name="single-file-upload-input"
+      />
+    </Box>
+  );
+};
+
+TestOnRemove.play = async () => {
+  const { getByText, getByRole, queryByText } = within(document.body);
+
+  // Create a file
+  const blob = new Blob([
+    'data:image/png;base64,R0lGODlhDwAPAKECAAAAzMzM/////wAAACwAAAAADwAPAAACIISPeQHsrZ5ModrLlN48CXF8m2iQ3YmmKqVlRtW4MLwWACH+H09wdGltaXplZCBieSBVbGVhZCBTbWFydFNhdmVyIQAAOw==',
+  ]);
+  const filename = 'my-image.png';
+  const file = new File([blob], filename, {
+    type: 'image/png',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
```

Comment:
```suggestion
  const input = getByRole('input', { name: 'single-file-upload-input' });
```

Will this work ^^?

Reply:
I tried that first, but it doesn't work for unknown reasons.